### DISCUSSION TITLE
SpringBoot Backend 2

### DISCUSSION THREAD

**Discussion by AntonIV8** [Flexberry/rfcs#30] [General]
Развитие существующего примера SpringBoot бэкенда

https://github.com/Flexberry/Flexberry.SpringReactApplication.Sample

Добавление в него части компонентов и сервисов, используемых в JHipster

**Comment by AntonIV8** [comment]
Перечень планируемых изменений и нововведений

1) Добавление в схему взаимодействия отдельного сервиса GrafanaLoki. В виде отдельного docker-контейнера
2) Добавление сервиса ElasticSearch для сбора метрик бэкенда. В виде отдельного докер-контейнера.
3) В самом SpringBoot бэкенде смена Web-сервера со стандартного Tomcat на Undertow
4) Добавление в схему взаимодействия сервиса-шины Kafka. В виде отдельного docker-контейнера
5) В качестве демонстрации работы Kafka, организовать через нее отправку логов бэкенда в сервис GrafanaLoki и метрик в ElasticSearch
6) Добавление в бэкенд HIkariCP - пул соединений с БД.
7) Добавление сервиса-прокси Spring Cloud Gateway. В виде отдельного докер-контейнера. Все запросы извне будут ходить через него (к интерфейсу эластика, графаны и бэкенду для получения данных)

**Comment by AntonIV8** [comment]
Схема

![image](https://github.com/Flexberry/rfcs/assets/13151962/ddfd70cb-3b1e-446f-8479-4edf2aeb6c0f)

**Comment by archaim** [comment]
В этой или следующей итерации рассмотрите возможность использования Camel, если ещё не. Через него намного проще налаживать цепочки взаимодействия, особенно, если используется шина.
Также, будет проще накатить метрики и централизованное логирование.

**Reply by AntonIV8** [reply]
Я не заметил его в JHipster

**Reply by bratchikov** [reply]
jHipster - это лишь один из источников сведений о лучших практиках, но он не единственный.
